instance/models/instance.py

```diff
@@ -586,4 +669,19 @@ def provision(self):
 
         return (server, ansible_log)
 
+    def provision_external_databases(self):
+        """
+        Provision external databases for this instance
+        """
+        if settings.INSTANCE_MYSQL_URL_OBJ:
+            connection = mysql.connect(
+                host=settings.INSTANCE_MYSQL_URL_OBJ.hostname,
+                user=settings.INSTANCE_MYSQL_URL_OBJ.username,
+                passwd=settings.INSTANCE_MYSQL_URL_OBJ.password or '',
+                port=settings.INSTANCE_MYSQL_URL_OBJ.port or 3306,
+            )
+            cursor = connection.cursor()
+            database_name = connection.escape_string(self.database_name).decode()
+            cursor.execute('CREATE DATABASE IF NOT EXISTS {0}'.format(database_name))
```

Comment:
I think you can leave escaping the DB name to the database driver – something like

    cursor.execute('CREATE DATABASE IF NOT EXISTS %s', [self.database_name])

or similar.  Not sure whether that works in this particular case, and I haven't checked the details.